src/griffe_warnings_deprecated/extension.py

```diff
@@ -15,8 +16,9 @@
 
 def _deprecated(obj: Class | Function) -> str | None:
     for decorator in obj.decorators:
-        if decorator.callable_path in _decorators:
-            return str(decorator.value).split("(", 1)[1].rstrip(")").rsplit(",", 1)[0].lstrip("f")[1:-1]
+        if decorator.callable_path in _decorators and isinstance(decorator.value, ExprCall):
+            message = str(decorator.value.arguments[0]).removeprefix("f")
+            return ast.literal_eval(message)
```

Comment:
OK if we want to be robust we could also check that `arguments[0]` is a string and nothing else (not even an f-string), because only static strings will be supported, at least statically.

```suggestion
            first_arg = decorator.value.arguments[0]
            try:
                return ast.literal_eval(first_arg)
            except ValueError:
                logger.debug("%s is not a static string", first_arg)
                return None
```

WDYT?

Reply:
I like it. I took the liberty of changing `repr(...)` to `str(...)`, since the logging message would otherwise read `ExprJoinedStr(values=['message']) is not a static string`, which was not the input. With `str(...)` it reads `f'message' is not a static string`.